packages/e2e-test-utils-playwright/src/admin/visit-post-editor.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * WordPress dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import type { Admin } from './';
+
+interface PostOptions {
+	postId?: string | number;
+	showWelcomeGuide?: boolean;
+}
+
+/**
+ * Creates new post.
+ *
+ * @param this
+ * @param options Options to create new post.
+ */
+export async function visitPostEditor(
+	this: Admin,
+	options: PostOptions = {}
+) {
+	if ( typeof options.postId === 'undefined' ) {
```

Comment:
Now that we have the opportunity to refactor this, I wonder if we should just combine `createNewPost` and `visitPostEditor`?

```js
// Call with `postId`: visit the existing post.
admin.visitPostEditor( postId: number ): Promise<void>;

// No `postId`: visit `post-new.php` instead.
admin.visitPostEditor(): Promise<void>;
```

WDYT?

Reply:
A few notes on this:

1. The https://github.com/WordPress/gutenberg/commit/59720fe6b5072c37063c43887339ecd2653ca003 is a breaking change. We shouldn't ship that as `@wordpress` packages try to follow the same rules as WP regarding backward compatibility.
2. Personally, I prefer a more explicit `createNewPost` action.

@WunderBart, I guess we can make `createNewPost` a wrapper around `visitPostEditor`. We just need to make sure that the argument signature is the same.

Reply:
The wrapper idea makes sense to me. cc @kevin940726 if he has any objections here!

Reply:
> Personally, I prefer a more explicit createNewPost action.

I'm actually starting to lean in this direction as well. The main issue with merging createNewPost and visitPostEditor are the props (and their typing) that can be confusing (e.g. if postId is passed, all the props for creating a new post are ignored). Personally, at this point I also feel like a more explicit approach would be better - keep the `createNewPost` and `editPost` instead of `visitPostEditor`. For the Site Editor, it makes sense to be reachable by `visitSiteEditor` because there's only one URL we can visit it by: /site-editor.php. 

Sorry for going back and forth with it, but I just want to avoid overcomplicating stuff. 😅 Please let me know if the above would make sense to you!

Reply:
> keep the `createNewPost` and implement `editPost` instead of `visitPostEditor`.

100% - I think this fits the WP glossary around posts much better.

> Sorry for going back and forth with it, but I just want to avoid overcomplicating stuff

No worries. It's easier to anticipate what would work best when we try. Thanks for taking the time and experimenting with different approaches.

Reply:
I appreciate the discussions and I agree with the final solution we settled on too! Thank you all 🧡